Add possibility to build GNU IDE Plugins as part of GNU Toolchain build

apolyakov opened the pull request.

simonpcook committed:
Create arc-versions.sh for arc-4_8-toolchain-stable

jeremybennett committed:
Based on an idea from Christian Ruppert
	<christian.ruppert@abilis.com>

	* build-all.sh: Add --uclibc-defconfig option, setting
	UCLIBC_DEFCFG environment variable.
	* build-uclibc.sh: Use UCLIBC_DEFCFG environment variable to
	specify defconfig for uClibc.

fbedard committed:
Update README.md

simonpcook committed:
* README.md: Update description of stable branches.

simonpcook committed:
* arc-versions.sh: Update branch names to new naming convention.

jeremybennett committed:
Fix problem with uClibc version.

	* arc-versions.sh: Correct uClibc version.

anthony-kolesov committed:
Mention that Git tags should be used to obtain stable releases

Also remove duplicate commands, fix typos and GFM syntax errors.

(cherry picked from commit abc32c27143eecc322ab5ccee8288e792cd3b432)

anthony-kolesov committed:
Fix error when build elf32-only toolchain with --rel-rpaths

Script rel-rpaths.sh was trying to update uClibc-specific libc.so file
unconditionally. As a result elf32-only toolchain build was failing. This
patch adds a check that whether libc.so file actually exists before trying
to update it.

Signed-off-by: Anton Kolesov <akolesov@synopsys.com>

anthony-kolesov committed:
Add a Makefile to create prebuilt packages automatically

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Bump Linux branch to arc-3.15

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Create arc-versions.sh for tag arc-4.8-R4-rc1

anthony-kolesov committed:
Use RHEL Mingw by default

Ubuntu defines MinGW toolchain as i686-w64-mingw32, while RHEL as
i686-pc-mingw32. This patch makes RHEL variant default in the build-elf32.sh
patch for Windows toolchain.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Update readme to current state of tools

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Rename EM to arcem and HS to archs in build scripts

GCC option has been renamed to arcem and archs, so build scripts should
match it.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Merge arc-4.8-dev into arc-4.8-stable

Conflicts:
	ChangeLog
	README.md
	arc-versions.sh
	build-all.sh
	build-uclibc.sh
	rel-rpaths.sh

anthony-kolesov committed:
Merge branch arc-4.8-dev into arc-4.8-stable

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2014.08-rc1

anthony-kolesov committed:
Merge commit '04f5559' into arc-4.8-stable

anthony-kolesov committed:
Use stable branches for release tags

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Merge branch arc-4.8-dev into arc-4.8-stable

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2014.08-rc2

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2014.08

anthony-kolesov committed:
Merge branch arc-4.8-dev into arc-4.8-stable

anthony-kolesov committed:
Merge branch 'arc-4.8-dev' into arc-releases

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2014.12-rc1

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2014.12-rc2

anthony-kolesov committed:
Build Linux toolchain for both ARC 700 and HS

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Merge branch 'arc-4.8-dev' into arc-releases

anthony-kolesov committed:
Merge branch 'arc-4.8-dev' into arc-releases

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2014.12-rc3

anthony-kolesov committed:
Merge branch 'arc-4.8-dev' into arc-releases

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2014.12-rc4

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2014.12

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-releases

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2015.06-rc1

anthony-kolesov committed:
Fix ncurses build for big endian

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Push arc-staging branch in push-release

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2015.06-rc2

anthony-kolesov committed:
Fix invalid linux tag in checkout config file

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Create 2015.06 release

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Release: Remove extra NOTPARALLEL targets

Ordering of targets is already tracked via dependencies. Besides NOTPARALLEL
was used improperly, because it makes sequential *all* targets after it, not
only those mentiond in the "dependency" list.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2015.12-rc1

anthony-kolesov committed:
Merge remote-tracking branch 'origin/arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2015.12

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.03-rc2

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.03

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-eng004

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Release: Omit -release suffix when creating final release tarballs

When release is created only the `arc-` prefix is stripped from the Git tag
name, so for tag arc-2016.09-eng005, filenames include 2016.09-eng005 in
them. This is not good for final releases, because Git tag is
arc-2016.09-release, so files will include 2016.09-release. This patch adds
a code that will remove `-release` from filenames part.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Build multi-OS ABI GDB

New GDB can support multiple targets and multiple OS ABI's at once. This
patch changes build configuration, so that baremetal GDB is built to support
linux applications as well. Default OS ABI is still the one for the
toolchain being built - proper OS/ABI is selected based on ELF file
sections.

Note that this is implemented only for baremetal GDB. The problem is that
baremetal OS ABI in GDB is "none" which is equal to "none", and due to
details of implementation in generic GDB code, GDB that is has "none" OS ABI
by default can switch to "GNU/Linux", but it cannot switch otherwise.
Possible solution is to use "Newlib" OS ABI for baremetal applications,
however then ARC GDB will need a custom OS ABI sniffer.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Fix an syntax error in Eclipse invocation

Spaces are not allowed when listing Eclipse plugins.

For some reasons `eclipsec.exe` doesn't work with Mars.2, use `eclipse.exe`
instead.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Add mention of wiki page about --specs option

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Check for Tex version in build-all.sh

Both build-elf32.sh and build-all.sh used to contain same code that checks
for Tex version and sets variable LC_ALL accordingly. It makes sense to have
this check in build-all.sh to avoid code repetition. In the process I've
also modified the way how version of Tex is checked - now version numbers
are exported in environment variables, so they can be reused in
build-elf32.sh and build-uclibc.sh scripts later.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Return 1 instead of exiting from build functions

Doing `exit 1` causes an instant fail, however in some cases we want to
handle make invocation error or just ignore it, hence `return 1` should be
used instead. Since we use `set -x` and `set -o pipefail` in most cases
there is no different between those two.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Build newlib pdf multiple times Texinfo >= 6.1

There is some issue with building newlib PDFs on Ubuntu 16.04. It might be
same as:

https://bugs.launchpad.net/ubuntu/+source/texlive-bin/+bug/1270396

PDF file is actually created, but exit code is 1, so it is not installed. As
a result we have to call make three times: one to build libc.pdf, one to
build libm.pdf and one to install them.

I assume that this is specific to Texinfo version 6.1 instead of being just
Ubuntu 16.04, so check is done against texinfo version.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Remove workaround for a fixed C++ bug

STAR 9000908736 has been fixed long ago, so there is no need to pass
--without-cxx-bindings when configuring ncurses.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Release: Fix build with disabled OpenOCD

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Release: Allow disabling of big endian toolchain build

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Release: Fix md5sum target when IDE is disabled

Target for md5sum files has been failing when ENABLE_WINDOWS_INSTALLER was
set to `n`, because installer is a unique target, built on a separate
machine outside of the `release.mk, so it doesn't use some of the common
variables, like `BUILD_DEPS`.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Do not build native GDB with C++ compiler

There is a bug somewhere, that causes a segmentation fault in GDB (see
STAR 9001066513). To avoid an error native GDB should be built with C
compiler, not a C++ one.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-eng006

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-eng007

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-eng008

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-eng009

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-eng010

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Revert "Add gitlab-ci.yml"

This reverts commit 874071099b9e8a0d49d5d3ac6202f1024acbdd7c.

anthony-kolesov committed:
Fix sporadic build error when installing libssp headers

It looks like that libssp install target is not parallel-friendly - I had
occassional issues, when installing it's header. I'm not entirely sure that
this is the case, but that is my best guess so far, considering that I
wasn't able to see any particular pattern of when it does happen. Error
message in log file is:

/usr/bin/install: cannot create regular file
`/home/arc/arc_gnu/toolchain/../INSTALL/lib/gcc/arc-elf32/6.1.1/include/ssp/stdio.h':
File exists

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-eng011

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-eng012

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-eng013

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-eng014

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-eng015

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-eng016

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-eng017

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-eng018

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-eng019

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-rc1

anthony-kolesov committed:
Add release notes for 2016.09 release

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Add one more entry to release notes for 2016.09

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
doc: Update Linux guide for 2016.09 release

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Mention that toolchain will be supported in next embARC release

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Readme: Fix missing directory name in Linux clone

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Fix typo in error message

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Accept TABs in gcc/config/arc/arc-cpus.def

File gcc/gcc/config/arc/arc-cpus.def is used to determine valid CPU models
for this GCC version. However, this stopped working with a recent commit to
GCC where tabs were added to this file and previous AWK field separator
haven't treated them properly.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Properly check IS_CROSS_COMPILING variable

Two checks for IS_CROSS_COMPILING value in build-elf32.sh assumed that this
variable is "true" if set, which is wrong, because build-all.sh always sets
this variable, either to "yes" or "no". This patch fixes this problem, now
checks is done against "yes" value of this variable to figure out of this is
a cross-compilation of toolchain.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-rc2

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-rc2

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-release

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2016.09-release

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2017.03-eng001

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2017.03-eng002

anthony-kolesov committed:
Build uClibc in source tree by default

I observe that it is relatively easy to trigger "argument list too long"
error when building uClibc out of the build tree, so it makes sense to
building in tree by default.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2017.03-eng003

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Remove engineering builds and RCs from configuration files

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2017.03-eng004

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2017.03-eng005

anthony-kolesov committed:
Release: Always use "kent" mirror of Sourceforge

This is a patch specific to Synopsys infrastructure, which doesn't allow
certain Sourceforge mirrors, therefore to ensure that builds always pass
this patches directly specifies allowed mirror.

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Create arc-versions.sh for tag arc-2017.03-eng006

anthony-kolesov committed:
Revert "Update Eclipse and JRE versions"

This reverts commit 498cc78eef0399eb9aacef9da940e7ed0e79e586.

anthony-kolesov committed:
Merge branch 'arc-dev' into arc-staging

anthony-kolesov committed:
Release: Update Java version to 8u121

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

Commit:
Create arc-versions.sh for tag arc-2017.03-eng008

Commit:
Merge branch 'arc-dev' into arc-staging

Commit:
Merge branch 'arc-dev' into arc-staging

Commit:
Create arc-versions.sh for tag arc-2017.03-eng009

Commit:
Create arc-versions.sh for tag arc-2017.03-eng010

Commit:
Merge branch 'arc-dev' into arc-staging

Commit:
Create arc-versions.sh for tag arc-2017.03-eng011

Commit:
Merge branch 'arc-dev' into arc-staging

Commit:
Create arc-versions.sh for tag arc-2017.03-eng012

Commit:
Merge branch 'arc-dev' into arc-staging

Commit:
Create arc-versions.sh for tag arc-2017.03-eng013

Commit:
Merge branch 'arc-dev' into arc-staging

Commit:
Merge branch 'arc-dev' into arc-staging

Commit:
Create arc-versions.sh for tag arc-2017.03-rc1

Commit:
Merge branch 'arc-dev' into arc-staging

Commit:
Create arc-versions.sh for tag arc-2017.03-rc2

Commit:
Create arc-versions.sh for tag arc-2017.03-release

anthony-kolesov committed:
Merge remote-tracking branch 'origin/arc-staging' into arc-releases

Signed-off-by: Anton Kolesov <Anton.Kolesov@synopsys.com>

apolyakov committed:
Add possibility to build GNU IDE Plugins as part of GNU Toolchain build

anthony-kolesov commented on release.mk:
```diff
@@ -457,8 +460,10 @@ ifeq ($(THIRD_PARTY_SOFTWARE_LOCATION),)
 endif
 
-	# Copy IDE plugin
+ifeq ($(ENABLE_IDE_PLUGINS_BUILD),)
```
Check should be `ifneq ($(ENABLE_IDE_PLUGINS_BUILD),y)` so `y` is the only value that means "yes" and any other value means "no".

anthony-kolesov commented on release.mk:
```diff
@@ -751,6 +756,21 @@ $O/$(ECLIPSE_VANILLA_LINUX_TGZ):
+OUTPUT_DIR=arc_gnu_eclipse/repository/target/
+
+$O/$(IDE_PLUGINS_ZIP):
+	$(call git_clone,arc_gnu_eclipse,arc_gnu_eclipse)
```
This should be in the `clone` target.

anthony-kolesov commented on release.mk:
```diff
@@ -751,6 +756,21 @@ $O/$(ECLIPSE_VANILLA_LINUX_TGZ):
+ifeq ($(ENABLE_IDE_PLUGINS_BUILD),y)
+
+BUILD_DIR=arc_gnu_eclipse/build
+OUTPUT_DIR=arc_gnu_eclipse/repository/target/
```
Those variables names are too ambigous and might clash with other compnents. They also should be declated close other variables, here https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/blob/arc-dev/release.mk#L299

You'd need something like this:
```Makefile
IDE_PLUGIN_SRC_DIR=$(ROOT)/arc_gnu_eclipse
IDE_PLUGIN_BUILD_DIR=$(IDE_PLUGIN_SRC_DIR)/build
IDE_PLUGIN_OUT_DIR=$(IDE_PLUGIN_SRC_DIR)/build/repository/target/
```

Although BUILD_DIR and OUTDIR, might not be useful at all, because each is used only once.

anthony-kolesov commented on release.mk:
```diff
@@ -39,6 +39,9 @@ ENABLE_DOCS_PACKAGE := n
 
+# Whether to build or download GNU IDE Plugins
+ENABLE_IDE_PLUGINS_BUILD := y
+
```
Please document new option in https://raw.githubusercontent.com/foss-for-synopsys-dwc-arc-processors/toolchain/arc-dev/doc/maintainer/release.rst

anthony-kolesov reviewed (changes requested):
You also need to change value of `IDE_PLUGINS_ZIP`, because it now needs to be arc_gnu_$(RELEASE)_ide_plugins.zip instead of `RELEASE_BRANCH`. However when you are copying prebuilt zip, it's name it still old-style, so you also has to change copy-external - source files uses RELEASE_BRANCH, but destination file uses RELEASE. This can be done in a separate commit.

apolyakov committed:
Change value of IDE_PLUGINS_ZIP

Since IDE plugins may be built as part of toochain, their name must be in
release output format

anthony-kolesov commented:
Duplicate of another PR.
